pkg/services/ngalert/state/historian/otel_loki_client.go

```diff
@@ -0,0 +1,184 @@
+package historian
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/metrics"
+	"github.com/unknwon/log"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+var _ remoteLokiClient = (*otelLokiClient)(nil)
+
+type otelLokiClient struct {
+	client  plogotlp.GRPCClient
+	once    *sync.Once
+	cfg     OtelConfig
+	metrics *metrics.Historian
+}
+
+func NewOtelLokiClient(cfg OtelConfig, metrics *metrics.Historian) *otelLokiClient {
+	return &otelLokiClient{
+		once:    &sync.Once{},
+		cfg:     cfg,
+		metrics: metrics,
+	}
+}
+
+func (p *otelLokiClient) Ping(context.Context) error {
+	return nil
+}
+
+func (p *otelLokiClient) RangeQuery(ctx context.Context, logQL string, start, end, limit int64) (QueryRes, error) {
+	return QueryRes{}, fmt.Errorf("unsupported operation")
+}
+
+func (p *otelLokiClient) Push(ctx context.Context, s []Stream) (err error) {
+	const (
+		timerFailureCode = "500"
+		exportMethodName = "OtelExport"
+	)
+	p.once.Do(func() {
```

Comment:
for clarity, let's put this in another function (`initClientConn` for example)

Reply:
changed to getClient with singleton in it